### PR TITLE
Release 0.5.2

### PR DESCRIPTION
**Pull request by klochowicz**
This PR will be merged back into master with the changelog after patch release.
Here, I just want to show what's included in release so I can get ACK from someone :)

### CODE REVIEW

**Comment by klochowicz** [comment]
I started it from 0.5.1, that's why it shows more than these 2 last commits (@Restioson 's change and my changelog update).

**Comment by da-kami** [comment]
> I started it from 0.5.1, that's why it shows more than these 2 last commits (@Restioson 's change and my changelog update).

OK, because you point it against `master`, got it.

**Comment by klochowicz** [comment]
Yes, I'm sure, 0.5.1 was not merged back to master after releasing :/ the last common ancestor from master is 0.5.0.

**Comment by da-kami** [comment]
I am a bit confused what the process is now - are you actually planning on merging this into master?
Would that trigger a release by the action then?

Usually we create patch releases manually.

**Comment by klochowicz** [comment]
> I am a bit confused what the process is now - are you actually planning on merging this into master? Would that trigger a release by the action then?
> 
> Usually we create patch releases manually.

we do them manually, but we should subsequently merge them back into master, so that changelog stays updated and we can reach all the tags from master branch.

**Comment by klochowicz** [comment]
I have created this PR right now just for visibility, so you can ACK the scope of the change, as written in the description. Can you ACK the scope of the change please? :-)

**Comment by da-kami** [comment]
ack the scope

**Comment by bonomat** [comment]
> we do them manually, but we should subsequently merge them back into master, so that changelog stays updated and we can reach all the tags from master branch.

My usual approach is to 
- create a branch for the new release
- cherry-pick commits from `master` 
- update changelog on new release branch
- trigger release from release branch
- create a PR against `master` with only the changelog change. 

Nothing else needs to be merged back into master.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Ignore frequent trace spans
+
```
Comment (bonomat) [inline CHANGELOG.md:13]: You will also need to update the end of this file where it says

```
[Unreleased]: https://github.com/itchysats/itchysats/compare/0.5.1...HEAD
[0.5.0]: https://github.com/itchysats/itchysats/compare/0.5.0...0.5.1
```

Note there was a typo which was fixed on master.

**Comment by klochowicz** [comment]
> > we do them manually, but we should subsequently merge them back into master, so that changelog stays updated and we can reach all the tags from master branch.
> 
> My usual approach is to
> 
>     * create a branch for the new release
> 
>     * cherry-pick commits from `master`
> 
>     * update changelog on new release branch
> 
>     * trigger release from release branch
> 
>     * create a PR against `master` with only the changelog change.
> 
> 
> Nothing else needs to be merged back into master.



> > we do them manually, but we should subsequently merge them back into master, so that changelog stays updated and we can reach all the tags from master branch.
> 
> My usual approach is to
> 
>     * create a branch for the new release
> 
>     * cherry-pick commits from `master`
> 
>     * update changelog on new release branch
> 
>     * trigger release from release branch
> 
>     * create a PR against `master` with only the changelog change.
> 
> 
> Nothing else needs to be merged back into master.

IFAICT the standard approach is to merge back the whole thing, as then you know for sure you have never missed anything important back on master and the tag is mainlined too. It also is easier, because you don't need to do additional cherry-pick of the changelog fixes. are there any downsides to that?
Even in gitflow hotfix release workflow (and our patch release is pretty much the same), this is the last step: https://medium.com/hard-work/gitflow-release-hotfix-bddee96fc5c3

Anyway, thanks for the writing down the steps you did, I'll incorporate it into the release process document, so we can continue this discussion there!

**Pull request closed without merging**